### ISSUE TITLE
The vault password file was not found

### ISSUE THREAD

**Issue by ReenigneArcher** [ansible/ansible-lint#3703]
##### Summary

Ansible lint gives warnings/errors that it cannot find the vault password file. I create my vault password using the following step before the ansible lint step.

```yml
run: echo ${{ secrets.MYSECRET || '1234' }} > .vault-password
```

I run ansible lint with this step:

```yml
uses: ansible/ansible-lint@v6.18.0
with:
  args: "playbooks/"
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```

6.18.0 (using ansible-lint as github action)

- ansible installation method: pypi
- ansible-lint installation method: using github action

##### STEPS TO REPRODUCE

Set password file location in config.

**ansible.cfg**
```ini
vault_password_file=.vault-password
```

Create vault passowrd file in workflow, as shown above.

Run ansible lint action, as shown above.

##### Desired Behavior

Find the file, that was created.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

File was not found.

```txt
Warning: /some_playbook.yml:1 [WARNING]: Error getting vault password file (default): The vault password file /runner/_work/ansible-repo/ansible-repo/.vault-password was not found
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Additional Information**

I use the same method of creating the .vault-password file to actually run ansible playbooks inside of GitHub actions, and that works fine. I also did not have this problem on the now depreciated `ansible-lint-action` repository.

**Comment by mszumilak** [comment]
This seems to be a duplicate of https://github.com/ansible/ansible-lint/issues/2443

**Comment by ReenigneArcher** [comment]
@mszumilak Maybe similar but I don't think it's a duplicate. They are having jinja and variable issues. I am having the issue where it cannot find the vault password file that does exist.

**Comment by ssbarnea** [comment]
@ReenigneArcher I do think that the vault password is not found because the current directory is different when running from lint. I would recommend you to ensure that you configure the path to vault file to be a full path, just in case.

Print the full of your dumped file, to check it.

**Comment by ReenigneArcher** [comment]
Specifying a full path to the password file is not the most practical advice as when you deploy the playbooks, the environments/directory structure likely don't match what you get with a GitHub runner.

Anyway, the password file exists at this location, which ansible lint is saying doesn't exist.

`/runner/_work/ansible-repo/ansible-repo/.vault-password`

**Comment by tamiroh** [comment]
I also encountered the same problem.

I think the reason for this is that actions/checkout deletes existing files when it runs. I think this is a specification and maybe  impossible to work around.

**Comment by ssbarnea** [comment]
How about dumping the file outside the cloning area, maybe even in TMP. This should work. I will convert the issue into a discussion as it does not seem like a bug.
